nep-11.mediawiki

```diff
@@ -0,0 +1,256 @@
+
+The parameter <code>tokenid</code> SHOULD be a valid NFT.
+If not, this method SHOULD <code>throw</code> an exception.
+
+The parameter <code>owner</code> SHOULD be a 20-byte address.
+If not, this method SHOULD <code>throw</code> an exception.
+
+If the <code>owner</code> is an unused address, or it's not the owner of the specified token, this method SHOULD return <code>0</code>.
+
+====tokensOf====
+
+<pre>
+public static enumerator tokensOf(byte[] owner)
+</pre>
+
+Returns an <code>enumerator</code> that contains all the tokens owned by the specified address.
+
+The parameter <code>owner</code> SHOULD be a 20-byte address.
+If not, this method SHOULD <code>throw</code> an exception.
+
+The returned tokens should be in the following format:
+<pre>
+{
+  "id": <ID>,
+  "uri": tokenURI(<ID>),
+  "properties": properties(<ID>),
+  ...
+}
+</pre>
+
+It must have the fields <code>id</code>, <code>uri</code> and <code>properties</code>.
+It can contain other dApp-specific fields.
+
+====tokenURI====
+
+<pre>
+public static string tokenURI(byte[] tokenid)
+</pre>
+
+Returns a distinct Uniform Resource Identifier (URI) for a given asset.
+URIs are defined in [https://tools.ietf.org/html/rfc3986 RFC 3986].
+This URI contains optionally mutable properties for the given NFT.
+
+The parameter <code>tokenid</code> SHOULD be a valid NFT.
+If not, this method SHOULD <code>throw</code> an exception.
+
+The URI MUST point to a JSON file that conforms to the "NEO NFT Metadata JSON Schema":
+
+The supported scheme of the returned URI MUST be one of the following: <code>http</code>, <code>https</code>, <code>data</code>.
+
+The "data" URI scheme is defined in [https://tools.ietf.org/html/rfc2397 RFC 2397].
+
+====properties====
+
+<pre>
+public static string properties(byte[] tokenid)
+</pre>
+
+Returns a serialized NVM object containing immutable on-chain properties for the given NFT.
+The NVM object must conform to the "NEO NFT Metadata JSON Schema".
+
+The parameter <code>tokenid</code> SHOULD be a valid NFT.
+If not, this method SHOULD <code>throw</code> an exception.
+
+====Metadata JSON Schema====
+
+<pre>
+{
+  "title": "Asset Metadata",
+  "type": "object",
+  "properties": {
+    "name": {
+      "type": "string",
+      "description": "Identifies the asset to which this NFT represents"
+    },
+    "description": {
+      "type": "string",
+      "description": "Describes the asset to which this NFT represents"
+    },
+    "image": {
+      "type": "string",
+      "description": "A URI pointing to a resource with mime type image/* representing the asset to which this NFT represents. Consider making any images at a width between 320 and 1080 pixels and aspect ratio between 1.91:1 and 4:5 inclusive."
+    }
+  }
+}
+</pre>
+
+The schema must contain a <code>title</code>, <code>type</code> and <code>properties</code>.
+
+Any custom fields can be included in the JSON file.
+Each field contains a <code>type</code> and <code>description</code>.
```

Comment:
And what's this?

Reply:
```
"name": {
    "type": "string",
    "description": "Identifies the asset to which this NFT represents"
}
```

You can see that `name` has `type` and `description`